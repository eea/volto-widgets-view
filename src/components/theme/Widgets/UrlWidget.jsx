import React from 'react';
import cx from 'classnames';

export const UrlWidget = ({ children, className }) => (
  <a
    href={children}
    className={cx(className, 'url', 'widget')}
    rel="noreferrer"
    target="_blank"
  >
    {children}
  </a>
);
