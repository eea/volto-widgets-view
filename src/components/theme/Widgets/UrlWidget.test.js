import React from 'react';
import renderer from 'react-test-renderer';
import { UrlWidget } from './UrlWidget';

describe('UrlWidget', () => {
  it('renders an URL view widget component', () => {
    const component = renderer.create(
      <UrlWidget className="metadata">http://foobar.com</UrlWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
