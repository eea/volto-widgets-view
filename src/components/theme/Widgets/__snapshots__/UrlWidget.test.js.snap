// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlWidget renders an URL view widget component 1`] = `
<a
  className="metadata url widget"
  href="http://foobar.com"
  rel="noreferrer"
  target="_blank"
>
  http://foobar.com
</a>
`;
